### PR TITLE
CLOUDP-66614: Enable linting on e2e tests

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ [Name of ticket](https://jira.mongodb.org/browse/[name-of-ticket])

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

## Further comments

I had a suspicious we weren't linting our e2e files but I could not pin the why, now I know

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -75,5 +75,15 @@ linters:
...
   skip-dirs:
     - internal/mocks
+
+issues:
+  # Excluding configuration per-path, per-linter, per-text and per-source
+  exclude-rules:
+    # Don't check for cyclomatic complexity on e2e file.
+    - path: e2e
+      linters:
+        - gocyclo
```
Comment (andreaangiolillo) [inline .golangci.yml:89]: [q] Could you explain me what this piece of code does?

Comment (andreaangiolillo) [reply .golangci.yml:89]: I am not sure what's the purpose of `issues`

Comment (gssbzn) [reply .golangci.yml:89]: what it says at the top
>  # Don't check for cyclomatic complexity on e2e file.

Our e2e files are long and complex and were hitting this lint I don't think there's value to fix them given that they read fine as they are and is an artefact of having to create something and then delete it in the same test

Comment (gssbzn) [reply .golangci.yml:89]: For clarity on `issues`, it's to ignore known issues on your code so you can add golint and not have to fix them all at once

**Pull request merged**